prow/cmd/webhook-server/validation.go

```diff
@@ -43,34 +43,35 @@ const (
 func (wa *webhookAgent) serveValidate(w http.ResponseWriter, r *http.Request) {
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		logrus.WithError(err).Fatal("Unable to read request")
+		logrus.WithError(err).Info("Unable to read request")
+		http.Error(w, fmt.Sprintf("bad request %v", err), http.StatusBadRequest)
 	}
 	admissionReview := &v1beta1.AdmissionReview{}
 	err = json.Unmarshal(body, admissionReview)
 	if err != nil {
-		logrus.WithError(err).Fatal("Unable to unmarshal admission review request")
+		logrus.WithError(err).Info("Unable to unmarshal admission review request")
```

Comment:
this is incorrect, there is no reason to run the rest of this function when the body is not unmarshallable, we should let the caller know and exit this function right away. Same with line 47, 58, 71, and all other occurrence of errors in this function as well as the function that serves mutating webhook

Reply:
to be more specific, when an error occurred serving a web request, normally you would:
- avoid crashing the server
- let the user know the most relevant error by `w.Error`
- stop serving that single request right away

so this line is more like 
```
logrus.WithError(err).Info("Unable to unmarshal admission review request")
w.Error(USER-FACING-ERROR, InternalErrorCode)
return
```